lib/global-styles.php

```diff
@@ -642,58 +716,19 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
  * @param array $config Config to extract values from.
  * @return array Default features config for the editor.
  */
-function gutenberg_experimental_global_styles_get_editor_features( $config ) {
-	$features = array();
+function gutenberg_experimental_global_styles_get_editor_settings( $config ) {
+	$settings = array();
 	foreach ( array_keys( $config ) as $context ) {
 		if (
-			empty( $config[ $context ]['features'] ) ||
-			! is_array( $config[ $context ]['features'] )
+			empty( $config[ $context ]['settings'] ) ||
+			! is_array( $config[ $context ]['settings'] )
 		) {
-			$features[ $context ] = array();
+			$settings[ $context ] = array();
 		} else {
-			$features[ $context ] = $config[ $context ]['features'];
+			$settings[ $context ] = $config[ $context ]['settings'];
 		}
 	}
-
-	// Deprecated theme supports.
```

Comment:
We were loading the old way of setting presets at the same time theme.json was loaded and the new features when retrieving the editor settings.

I think they should be loaded together as part of loading the theme theme.json.

The previous (master) implementation as an issue if the user overwrites one of these configurations using the user theme.json the theme supports continue to take precedence while they should not. As user configurations should overwrite theme configurations.